test/running_modules/run_tests

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python
+
+from __future__ import annotations
+
+import subprocess
+
+
+targets = [
+    ["--target", "erlang"],
+    ["--target", "javascript", "--runtime", "node"],
+    ["--target", "javascript", "--runtime", "deno"],
+]
+
+
+def main():
+    any_failed = not all([
+        test_command_output(
+            ["cargo", "run", "--", "run", "-m", "running_modules"],
+            expected="top level module",
+        ),
+        test_command_output(
+            ["cargo", "run", "--", "run", "-m", "running_modules/sub_module"],
+            expected="sub module",
+        ),
+    ])
+    raise SystemExit(any_failed)
+
+
+def test_command_output(args: list[str], *, expected: str) -> bool:
+    for target in targets:
+        args_copy = args.copy()
+        args_copy.extend(target)
+        print("Testing command ", args_copy, "...", sep="")
+        output = subprocess.run(args_copy, capture_output=True)
+
+        if expected not in output.stdout.decode("utf-8"):
+            print("Failed when running command", args)
+            print("with stdout:", output.stdout.decode("utf-8"))
+            return False
+
+        print("Passed")
+    return True
+
+
+if __name__ == "__main__":
+    main()
```

Comment:
This is cool but I would prefer to not introduce Python. Could we use shell?

```sh
should_succeed() {
  echo Running: "$@"
  eval "$@"
  if [ $? -ne 0 ]
  then
    echo ERROR: Command should have succeeded
    exit 1
  fi
}
 should_fail() {
  echo Running: "$@"
  eval "$@"
  if [ $? -eq 0 ]
  then
    echo ERROR: Command should have failed
    exit 1
  fi
 }

# No module
should_succeed gleam run --target erlang
should_succeed gleam run --target javascript --runtime nodejs
should_succeed gleam run --target javascript --runtime deno

# Module from same package
should_succeed gleam run --module one --target erlang
should_succeed gleam run --module one --target javascript --runtime nodejs
should_succeed gleam run --module one --target javascript --runtime deno

# Nested module from same package
should_succeed gleam run --module one/two --target erlang
should_succeed gleam run --module one/two --target javascript --runtime nodejs
should_succeed gleam run --module one/two --target javascript --runtime deno

# Nested module from another package
should_succeed gleam run --module dep/mod --target erlang
should_succeed gleam run --module dep/mod --target javascript --runtime nodejs
should_succeed gleam run --module dep/mod --target javascript --runtime deno

# Unknown module
should_fail gleam run --module wibble

# No main function
should_fail gleam run --module wobble

# Main function with wrong arity
should_fail gleam run --module wabble
```

Reply:
done. I didn't test a dependency since I don't have a dependency to test. Is there one i should use?

Reply:
Anything would work. How about gleeunit? That has a main function.

Reply:
I used gleeunit.